### PR TITLE
fix Json4sScalaModule deserialisation logic

### PR DESCRIPTION
**Pull request by magnolia-k**
https://github.com/json4s/json4s/issues/603

Fixed the problem that deserialization was not correctly done
when passing something other than JValue to the readValue method
(eg JObject).

### CODE REVIEW

**Comment by xuwei-k** [comment]
Could you add tests?

**Comment by magnolia-k** [comment]
Added tests!

File: tests/src/test/scala/org/json4s/JacksonDeserializationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.json4s
+
+import org.specs2.mutable.Specification
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.json4s.jackson.Json4sScalaModule
+import org.json4s._
```
Comment (xuwei-k) [inline tests/src/test/scala/org/json4s/JacksonDeserializationSpec.scala]: unused import

Comment (magnolia-k) [reply tests/src/test/scala/org/json4s/JacksonDeserializationSpec.scala]: oh...thanks for review
I removed

Comment (magnolia-k) [reply tests/src/test/scala/org/json4s/JacksonDeserializationSpec.scala]: oh...thanks for review
I removed

**Pull request merged**